src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -147,10 +147,17 @@ public void TrackNewActiveWindowFrame(IVsWindowFrame frame)
 
             _activeFrame = frame;
 
-            if (!_visibleFrames.Any(f => f.Frame == frame))
+            var existingFrame = _visibleFrames.FirstOrDefault(f => f.Frame == frame);
+            if (existingFrame == null)
             {
                 _visibleFrames = _visibleFrames.Add(new FrameListener(this, frame));
             }
+            else if (existingFrame.TextBuffer == null)
+            {
+                // If no text buffer is associated with existing frame, remove the existing frame and add the new one.
```

Comment:
Is this because there's some more async buffer creation now, and we're just looking for a text buffer too early?

Reply:
I am honestly not sure, but we do get multiple callbacks where sometimes getting the DocData property succeeds and sometimes it doesn't. If you'd like, I can create a separate issue for someone more familiar with this space (probably you?) to debug through this.

Reply:
OK to merge this, but I'd say reach out to the extensibility discussion internal list and see what they have to say.